pkg/capture/metrics.go

```diff
@@ -12,108 +12,217 @@ const (
-)
-var promGlobalBufferUsage = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-	Namespace: config.ServiceName,
-	Subsystem: captureSubsystem,
-	Name:      "global_packet_buffer_usage",
-	Help:      "Percentage of global buffer capacity used during flow map rotation",
-},
-	[]string{"iface"},
-)
-var promNumFlows = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-	Namespace: config.ServiceName,
-	Subsystem: captureSubsystem,
-	Name:      "flows_total",
-	Help:      "Number of flows tracked in the flow map",
-},
-	[]string{"iface"},
-)
-var promPacketsDropped = prometheus.NewCounterVec(prometheus.CounterOpts{
-	Namespace: config.ServiceName,
-	Subsystem: captureSubsystem,
-	Name:      "packets_dropped_total",
-	Help:      "Number of packets dropped",
-},
-	[]string{"iface"},
-)
-var promCaptureIssues = prometheus.NewCounterVec(prometheus.CounterOpts{
-	Namespace: config.ServiceName,
-	Subsystem: captureSubsystem,
-	Name:      "capture_issues_total",
-	Help:      "Number of unexpected issues encountered during packet capture",
-},
-	[]string{"iface", "issue_type"},
+var (
+	DefaultMetricsHistogramBins = []float64{0.01, 0.05, 0.1, 0.25, 1, 5, 10, 30, 60, 300}
 )
 
-var promInterfacesCapturing = prometheus.NewGauge(prometheus.GaugeOpts{
-	Namespace: config.ServiceName,
-	Subsystem: captureManagerSubsystem,
-	Name:      "interfaces_capturing_total",
-	Help:      "Number of interfaces that are actively capturing traffic",
-})
-
-// not exposing the interface due to the high-cardinality nature of the histogram
-var promRotationDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
-	Namespace: config.ServiceName,
-	Subsystem: captureManagerSubsystem,
-	Name:      "rotation_duration_seconds",
-	Help:      "Total flow map rotation time, aggregated across all interfaces",
-	// rotation is significantly faster than the writeout. Hence the small buckets
-	Buckets: []float64{0.001, 0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 0.75, 1},
-})
-
-func init() {
+type promMetrics struct {
+	promPacketsProcessed *prometheus.CounterVec
+	promBytes            *prometheus.CounterVec
+	promPackets          *prometheus.CounterVec
+	promPacketsDropped   *prometheus.CounterVec
+	promCaptureIssues    *prometheus.CounterVec
+
+	promGlobalBufferUsage *prometheus.GaugeVec
+	promNumFlows          *prometheus.GaugeVec
+
+	promInterfacesCapturing prometheus.Gauge
+	promRotationDuration    prometheus.Histogram
+}
+
+// Metrics denotes all capture-specific metrics, tracked in Prometheus
+type Metrics struct {
+	promMetrics
+
+	trackIfaces bool
+}
+
+// MetricsOption denotes a functional option for Metrics
+type MetricsOption func(*Metrics)
+
+// DisableIfaceTracking removes the "iface" label from all Prometheus
+// metrics (reducing cardinality, in particular if many interfaces from many sensors
+// are being tracked)
+func DisableIfaceTracking() MetricsOption {
+	return func(m *Metrics) {
+		m.trackIfaces = false
+	}
+}
+
+func NewMetrics(opts ...MetricsOption) *Metrics {
```

Comment:
Avoids ugly implicit handling via `init()` and makes it a proper, injectable type...

Reply:
Nit: comment missing